packages/matter.js/src/protocol/ControllerCommissioningHandler.ts

```diff
@@ -0,0 +1,783 @@
+        }
+
+        return {
+            code: CommissioningStepResultCode.Skipped,
+            breadcrumb: this.lastBreadcrumb,
+        }
+    }
+
+    private async configureNetworkWifi() {
+        if (this.commissioningOptions.wifiNetwork === undefined) {
+            logger.debug("Commissioning step 11: WiFi network is not configured");
+            return {
+                code: CommissioningStepResultCode.Skipped,
+                breadcrumb: this.lastBreadcrumb,
+            }
+        }
+        if (this.collectedCommissioningData.networkFeatures !== undefined && this.collectedCommissioningData.networkStatus !== undefined) {
+            const rootNetworkFeatures = this.collectedCommissioningData.networkFeatures.find(({ endpointId }) => endpointId === 0)?.value;
+            const rootNetworkStatus = this.collectedCommissioningData.networkStatus.find(({ endpointId }) => endpointId === 0)?.value;
+
+            logger.debug(`Root Networks found: ${Logger.toJSON(rootNetworkFeatures)} - ${Logger.toJSON(rootNetworkStatus)}`);
+
+            if (rootNetworkFeatures?.wiFiNetworkInterface !== true) {
+                logger.debug("Commissionee does not support any WiFi network interface");
+                return {
+                    code: CommissioningStepResultCode.Skipped,
+                    breadcrumb: this.lastBreadcrumb,
+                }
+            }
+            if (rootNetworkStatus !== undefined && rootNetworkStatus.length > 0 && rootNetworkStatus[0].connected !== false) {
+                logger.debug("Commissionee is already connected to the WiFi network");
+                return {
+                    code: CommissioningStepResultCode.Skipped,
+                    breadcrumb: this.lastBreadcrumb,
+                }
+            }
+        }
+        const networkCommissioningClusterClient = this.getClusterClient(NetworkCommissioning.Cluster.with("WiFiNetworkInterface"), 0, true);
+        const ssid = ByteArray.fromString(this.commissioningOptions.wifiNetwork.wifiSsid);
+        const credentials = ByteArray.fromString(this.commissioningOptions.wifiNetwork.wifiCredentials);
+
+        const { networkingStatus, wiFiScanResults, debugText } = await networkCommissioningClusterClient.scanNetworks({ ssid, breadcrumb: this.lastBreadcrumb++ });
+        if (networkingStatus !== NetworkCommissioning.NetworkCommissioningStatus.Success) {
+            throw new CommissioningError(`Commissionee failed to scan for WiFi networks: ${debugText}`);
+        }
+        if (wiFiScanResults === undefined || wiFiScanResults.length === 0) {
+            throw new CommissioningError(`Commissionee did not return any WiFi networks for the requested SSID ${this.commissioningOptions.wifiNetwork.wifiSsid}`);
+        }
+
+        const { networkingStatus: addNetworkingStatus, debugText: addDebugText, networkIndex } = await networkCommissioningClusterClient.addOrUpdateWiFiNetwork({ ssid, credentials, breadcrumb: this.lastBreadcrumb++ });
```

Comment:
Do we have anything in place in the lower levels to filter credentials from message logs?

Reply:
no, especially not when it is later "in the Tlv bytestream" ... I also have not ereally an idea how to prevent it there.